app/code/Magento/Checkout/Controller/Cart/CouponPost.php

```diff
@@ -95,44 +95,50 @@ public function execute()
                 if (!$itemsCount) {
                     if ($isCodeLengthValid && $coupon->getId()) {
                         $this->_checkoutSession->getQuote()->setCouponCode($couponCode)->save();
-                        $this->messageManager->addSuccess(
+                        $this->messageManager->addSuccessMessage(
                             __(
                                 'You used coupon code "%1".',
                                 $escaper->escapeHtml($couponCode)
                             )
                         );
                     } else {
-                        $this->messageManager->addError(
-                            __(
-                                'The coupon code "%1" is not valid.',
-                                $escaper->escapeHtml($couponCode)
-                            )
+                        $this->messageManager->addComplexErrorMessage(
```

Comment:
@adrian-martinez-interactiv4 would you explain why error message is created with `addComplexErrorMessage` but success message with simple `addSuccessMessage `. Messaging is pretty same so in both cases should be used same approach

Reply:
Yes, it seems so, it seems to be a mistake, but I don't remember if I did it this for some reason, I'm checking it